derp/derp.go

```diff
@@ -11,3 +11,31 @@
 // cannot be found or opened. DERP is a last resort. Both sides
 // between very aggressive NATs, firewalls, no IPv6, etc? Well, DERP.
 package derp
+
+import (
+	"encoding/binary"
+	"io"
+	"time"
+)
+
+const magic = 0x44c55250 // "DERP" with a non-ASCII high-bit
+
+type frameType byte
```

Comment:
Type and value docs? Unexported type so welcome to ignore, it's reasonably clear from context if none of the values have surprising implications.

Reply:
Done.

Reply:
(Well, not value docs... that was just movement.)

I'm still figuring out the protocol. :)